nav2_planner/src/planner_server.cpp

```diff
@@ -53,6 +54,7 @@ PlannerServer::PlannerServer(const rclcpp::NodeOptions & options)
   declare_parameter("planner_plugins", default_ids_);
   declare_parameter("expected_planner_frequency", 1.0);
   declare_parameter("costmap_update_timeout", 1.0);
+  declare_parameter("allow_partial_planning", true);
```

Comment:
As you mentioned, this needs to go into the migration guide for this to be merged.

I think also by default this should be set to `false` since I think alot of applications would not want partial paths. I think this could cause some user tickets for unexpected behavior.